src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp

```diff
@@ -50,94 +101,179 @@ TypeLookupRequestListener::~TypeLookupRequestListener()
 {
 }
 
+void TypeLookupRequestListener::check_get_types_request(
+        SampleIdentity request_id,
+        const TypeLookup_getTypes_In& request)
+{
+    TypeLookup_getTypes_Out out;
+    // Iterate through requested type_ids
+    for (const xtypes::TypeIdentifier& type_id : request.type_ids())
+    {
+        xtypes::TypeObject obj;
+        if (RETCODE_OK == fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                        get_type_object(type_id, obj))
+        {
+            xtypes::TypeIdentifierTypeObjectPair pair;
+            pair.type_identifier(type_id);
+            pair.type_object(obj);
+            // Add the pair to the result
+            out.types().push_back(std::move(pair));
+        }
```

Comment:
This is not correct with respect to the design decision of always using COMPLETE information by default. According with the specification:

* The kind of the TypeIdentifier should be checked. If it is COMPLETE, then the current return information would be correct with the design.
* But Fast DDS may be communicating with another vendor. XTypes specification does not enforce that the getTypeDependencies operation returns COMPLETE TypeIdentifiers. So MINIMAL TypeIdentifiers might be received. In this case, Fast DDS would answer with MINIMAL TypeObjects, when it was decided that Fast DDS would always send the COMPLETE information (unless the corresponding property is set in the Participant).
* Consequently, the logic should be extended to check the TypeIdentifier kind and in case of being a MINIMAL TypeId, send the Complete information and fill the `complete_to_minimal` field.
* It would be nice to enforce a test to check this case.
* Please, leave a TODO indicating that this would need to be refactored when the aforementioned property is implemented.

Reply:
added in [29681bc](https://github.com/eProsima/Fast-DDS/pull/4240/commits/29681bc3ecb5a0cf1f4e298b7bf093bff1ab1753)